pkg/proxy/parser.go

```diff
@@ -236,6 +236,12 @@ func (p *Parser) isEnterKeyPress(b []byte) bool {
 	if bytes.ContainsRune(b, '\r') {
 		return true
 	}
+	if p.TerminalParser != nil && p.TerminalParser.screenType == UsqlScreen {
+		// terminal 右键粘贴时，没有 \r 只有 \n
+		if bytes.ContainsRune(b, '\n') && bytes.ContainsRune(b, ';') {
+			return true
+		}
+	}
 	return false
 }
 
```

Comment:
It appears that there were no syntax errors, warnings, or potential changes to be observed. The code seems fine and doesn't contain any irregularities, possible issues, or optimizations at this moment. However, without specific context about the use case of this function, it's difficult to suggest improvements.

Would you like more detailed comments on how a parser might approach handling keypresses? This will help optimize parsing according to user behavior or ensure compatibility with other applications.